Source/SequenceMethodCall.cs

```diff
@@ -0,0 +1,97 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Linq.Expressions;
+using System.Reflection;
+
+using Moq.Proxy;
+
+namespace Moq
+{
+	// this corresponds to a setup created by `mock.SetupSequence`
+	internal class SequenceMethodCall : MethodCall
+	{
+		// contains the responses set up with `CallBase`, `Pass`, `Returns`, and `Throws`
+		// TODO: the use of a queue without keeping track of the number of received calls, and tying each
+		// response to an expected invocation count, means that things won't work quite like they used to!
+		// this is possibly a breaking change right now.
+		private ConcurrentQueue<Tuple<ResponseKind, object>> responses;
+
+		public SequenceMethodCall(Mock mock, Condition condition, Expression originalExpression, MethodInfo method, params Expression[] arguments)
+			: base(mock, condition, originalExpression, method, arguments)
+		{
+			this.responses = new ConcurrentQueue<Tuple<ResponseKind, object>>();
+		}
+
+		public void AddCallBase()
+		{
+			this.responses.Enqueue(Tuple.Create(ResponseKind.CallBase, (object)null));
+		}
+
+		public void AddPass()
+		{
+			this.responses.Enqueue(Tuple.Create(ResponseKind.Pass, (object)null));
+		}
+
+		public void AddReturns(object value)
+		{
+			this.responses.Enqueue(Tuple.Create(ResponseKind.Returns, value));
+		}
+
+		public void AddThrows(Exception exception)
+		{
+			this.responses.Enqueue(Tuple.Create(ResponseKind.Throws, (object)exception));
+		}
+
+		public override void Execute(ICallContext call)
+		{
+			base.Execute(call);
+
+			if (this.responses.TryDequeue(out Tuple<ResponseKind, object> response))
+			{
+				switch (response.Item1)
+				{
+					case ResponseKind.Pass:
+						break;
+
+					case ResponseKind.CallBase:
+						call.InvokeBase();
+						break;
+
+					case ResponseKind.Returns:
+						call.ReturnValue = response.Item2;
+						break;
+
+					case ResponseKind.Throws:
+						throw (Exception)response.Item2;
+				}
+			}
+			else
+			{
+				// we get here if there are more invocations than configured responses.
+				// if the setup method does not have a return value, we don't need to do anything;
+				// if it does have a return value, we produce the default value.
+
+				if (call.Method.ReturnType == typeof(void))
+				{
+
+				}
+				else if (call.Method.ReturnType.GetTypeInfo().IsValueType)
+				{
+					call.ReturnValue = Activator.CreateInstance(call.Method.ReturnType);
```

Comment:
AFAIK other parts of Moq do this in a more complicated fashion, because of nullable value types, COM types, etc. Check whether this is sufficient.